brainlit/algorithms/trace_analysis/fit_spline.py

```diff
@@ -144,9 +144,11 @@ def fit_spline_tree_invariant(self):
                 curr_spline_num, path=main_branch, starting_length=tree[2]
             )
             spline_tree.add_edge(parent_num, curr_spline_num)
-
-            for tree in collateral_branches:
-                stack.append((tree, curr_spline_num))
+            if len(collateral_branches) == 0 and len(stack) == 0:
```

Comment:
why did you make these changes? add info to PR

Reply:
Before the tested tree was extended in the pytest, these lines (148-149) were missed in the code coverage test. Then I tried to monitor the contents of ```collateral_branches``` and ```stack``` during the operation and found that the for loop would end when the length of these two variables become zeros. I hypothesized that the miss of executing the two lines occurred at the final round of the for loop where variables ended up empty arrays. Therefore, to make sure every line is hit, I added the if condition.
However, the problem was coming from the tested tree, which didn't have collateral branches.

Reply:
Ok, so sounds like you don't need the conditional anymore?

Reply:
The redundant condition [removed](https://github.com/neurodata/brainlit/blob/CTalgCodCov/brainlit/algorithms/trace_analysis/fit_spline.py#L148-L149).